### PR TITLE
feat: scheduler now can emit missed events

### PR DESCRIPTION
**Pull request by tribunsky-kir**

### CODE REVIEW

File: blocks/sources/schedule.py [hunk 1]
```diff
@@ -28,20 +28,30 @@ class Scheduler(Source):
...
 
+        self._skip_when_busy = skip_when_busy
+
     def __call__(self, _: Any = None) -> EventOrEvents:
-        if self._job.should_run:
-            self._job.last_run = datetime.now()
-            self._job._schedule_next_run()
-            return [self._event()]
+        if self._skip_when_busy:
+            if self._job.should_run:
```
Comment (tofuurem) [inline blocks/sources/schedule.py:41]: Syntactic sugar, but why not combine into 1 clause

File: blocks/sources/schedule.py [hunk 2]
```diff
@@ -28,20 +28,30 @@ class Scheduler(Source):
...
+                return self.__emit_single_event()
+        else:
+            if not self._job.last_run:
+                return self.__emit_single_event()
```
Comment (tofuurem) [inline blocks/sources/schedule.py:45]: Here on this line and one line below, a condition on the `self._job.last_run` attribute

By default, you always have it None: [schedule/__init__.py#L241](https://github.com/dbader/schedule/blob/010236b4d606940dc7682d2d736875dac1bd1375/schedule/__init__.py#L241)

And I don't really understand if it's empty you have a startup going, and yet the second condition will never work again.

But the logic itself is that in the first case it works, but in the second case it's a RuntimeException....

Am I missing something?

Comment (tribunsky-kir) [reply blocks/sources/schedule.py:45]: This API of `schedule`. The Job itself is mutable, initiated with optionals by default. But after the very first scheduling event `None` in fields will be replaced with actual values.

File: blocks/sources/schedule.py (hunk 1 above)
Comment (tribunsky-kir) [reply blocks/sources/schedule.py:41]: Cause it changes what will go to else ;)

Comment (tofuurem) [reply blocks/sources/schedule.py:41]: logically logical

File: blocks/sources/schedule.py (hunk 2 above)
Comment (tofuurem) [reply blocks/sources/schedule.py:45]: Well then like I said, after the first time, the second `if self._job.last_run is None` condition will never work, will it?

Comment (tribunsky-kir) [reply blocks/sources/schedule.py:45]: The whole else works like:
1. We accept that default schedule behavior with missing events is not that we want [optionally]
2. To not rewrite from scratch, we reuse the same job and API
3. When we run it first time job is initiated with `None`s
4. When we `__emit_single_event`, we set last_run manually and schedule's `_schedule_next_run` does all other dirty work. And yes, this clause is needed only once.
5. RuntimeError chick is necessary for mypy. Nuff said, it doesn't understand what will happen on the step 4
6. We manually calculate number of events that should befor passed time

**Review by tofuurem — APPROVED** [review]

**Pull request merged**